Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -965,20 +965,12 @@ public AndConstraint<TAssertions> ContainItemsAssignableTo<TExpectation>(string
 
             if (success)
             {
-                int index = 0;
-                foreach (T item in Subject)
-                {
-                    if (item is not TExpectation)
-                    {
-                        Execute.Assertion
-                            .BecauseOf(because, becauseArgs)
-                            .FailWith(
-                                "Expected {context:collection} to contain only items of type {0}{reason}" +
-                                ", but item {1} at index {2} is of type {3}.", typeof(TExpectation), item, index, item.GetType());
-                    }
-
-                    ++index;
-                }
+                var actualItems = Subject.ConvertOrCastToCollection();
+                Execute.Assertion
+                    .BecauseOf(because, becauseArgs)
+                    .WithExpectation("Expected {context:collection} to contain element assignable to type {0}{reason}, ", typeof(TExpectation).FullName)
```

Comment:
🔧 I would rephrase the middle of the message as "to contain at least one element...."

Reply:
Done in 3e7255089ab330db576c5c91d5a63daf8f10611a. Refactored method to contain a single assertion to remove _Expectation_'s message duplication.